### ISSUE TITLE
During training, the episode_length is always 1, and during play the robot is floating in the air

### ISSUE THREAD

**Issue by Edwardle06** [mujocolab/mjlab#403]
May I ask what might be causing this? The robot’s initial position and state are both on the ground.

**Comment by louislelay** [comment]
Hello @Edwardle06, we'll need more information to understand your issue! If you have some code, logs, better description of your problem, etc please share

**Comment by Edwardle06** [comment]
> Hello [@Edwardle06](https://github.com/Edwardle06), we'll need more information to understand your issue! If you have some code, logs, better description of your problem, etc please share

<img width="1441" height="961" alt="Image" src="https://github.com/user-attachments/assets/8f73af22-09a2-4d00-8f99-57784a012468" />

<img width="562" height="1115" alt="Image" src="https://github.com/user-attachments/assets/4cfd26f6-2256-48a9-85c3-9d96a7054f2e" />
These are my training and play information. From the play image, it can be seen that the robot's feet are not touching the ground. I suspect this is causing the episode length to remain at 1 during training, but I don't know where the problem lies.

**Comment by louislelay** [comment]
Could you share your asset code?

**Comment by Edwardle06** [comment]
[assets.xml](https://github.com/user-attachments/files/24117817/assets.xml)
[serial_actuators.xml](https://github.com/user-attachments/files/24117816/serial_actuators.xml)
[serial_links.xml](https://github.com/user-attachments/files/24117820/serial_links.xml)
[serial_pm_v2.xml](https://github.com/user-attachments/files/24117818/serial_pm_v2.xml)
[serial_sensors.xml](https://github.com/user-attachments/files/24117819/serial_sensors.xml)

[pm01_constants.py](https://github.com/user-attachments/files/24117836/pm01_constants.py) Do you mean this?

**Comment by louislelay** [comment]
Yes, but you did not gave me the right xml file, in your constants file you use this:

```
PM01_XML: Path = (
  MJLAB_SRC_PATH / "asset_zoo" / "robots" / "pm01" / "xmls"/"pm01.xml"
)
assert PM01_XML.exists()

```
So could you please send me the `pmo1.xml`?

**Comment by Edwardle06** [comment]
[pm01.xml](https://github.com/user-attachments/files/24117927/pm01.xml)

**Comment by louislelay** [comment]
Oh, okay... I expected your asset to be missing the freejoint, but you have it. Could you share what happens when you run:

```
simulate path/to/pm01.xml

uv run path/to/pm01_constants.py

uv run play Mjlab-Your-Task-Id --agent zero
```

**Comment by Edwardle06** [comment]
https://github.com/user-attachments/assets/8405e5a3-f193-4068-8c07-d237aca33cdc When I run "
uv run path/to/pm01_constants.py",we can see : https://github.com/user-attachments/assets/8405e5a3-f193-4068-8c07-d237aca33cdc 


But,when I run "simulate path/to/pm01.xml" and "
uv run Mjlab-Your-Task-Id --agent zero", there are something wrong:


 (mjlab) wangzhiwen@wangzhiwen-GeekPro-17IRR:~/mjlab-main/src/mjlab/scripts$ simulate /home/wangzhiwen/mjlab-main/src/mjlab/asset_zoo/robots/pm01/xmls/pm01.xml
simulate：未找到命令
(mjlab) wangzhiwen@wangzhiwen-GeekPro-17IRR:~/mjlab-main/src/mjlab/scripts$ uv run train Mjlab-Tracking-Flat-PM01-No-State-Estimation --agent zero
/home/wangzhiwen/mjlab-main/.venv/lib/python3.13/site-packages/tyro/_resolver.py:581: TyroWarning: <class 'list'> does not match any type in Union: [<class 'str'>, <class 'tuple'>, <class 'NoneType'>]
  warnings.warn(
╭─ Unrecognized options ───────────────────────────────────────────────────────────────────────────────────────────────────────────────╮
│ Unrecognized options: --agent, zero                                                                                                  │
│ ──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────── │
│ Perhaps you meant:                                                                                                                   │
│     --agent.algorithm.class-name STR                                                                                                 │
│         Ignore, required by RSL-RL. (default: PPO)                                                                                   │
│     --agent.algorithm.clip-param FLOAT                                                                                               │
│         The clipping parameter for the policy. (default: 0.2)                                                                        │
│     --agent.algorithm.desired-kl FLOAT                                                                                               │
│         The desired KL divergence between the new and old policies. (default: 0.01)                                                  │
│     --agent.algorithm.entropy-coef FLOAT                                                                                             │
│         The coefficient for the entropy loss. (default: 0.005)                                                                       │
│     --agent.algorithm.gamma FLOAT                                                                                                    │
│         The discount factor. (default: 0.99)                                                                                         │
│     --agent.algorithm.lam FLOAT                                                                                                      │
│         The lambda parameter for Generalized Advantage Estimation (GAE). (default: 0.95)                                             │
│     --agent.algorithm.learning-rate FLOAT                                                                                            │
│         The learning rate. (default: 1e-05)                                                                                          │
│     --agent.algorithm.max-grad-norm FLOAT                                                                                            │
│         The maximum gradient norm for the policy. (default: 1.0)                                                                     │
│     --agent.algorithm.normalize-advantage-per-mini-batch {True,False}                                                                │
│         Whether to normalize the advantage per mini-batch. Default is False. If True, the                                            │
│         advantage is normalized over the mini-batches only. Otherwise, the advantage is                                              │
│         normalized over the entire collected trajectories. (default: False)                                                          │
│     --agent.algorithm.num-learning-epochs INT                                                                                        │
│         The number of learning epochs per update. (default: 5)                                                                       │
│ ──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────── │
│ For full helptext, run /home/wangzhiwen/mjlab-main/.venv/bin/train Mjlab-Tracking-Flat-PM01-No-State-Estimation --help               │
╰──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────

**Comment by louislelay** [comment]
Okay, so for the the non working command, it's play, not train, and you need to add `--registry-name` as it's a motion tracking task

```
 uv run play Mjlab-Tracking-Flat-PM01-No-State-Estimation --agent zero --registry-name your-org/motions/motion-name
```

**Comment by Edwardle06** [comment]
Okay, this is what happened.

https://github.com/user-attachments/assets/f0bc0c26-e004-4e84-b47f-14101ae50c85

**Comment by louislelay** [comment]
Looking at that it's most probably from your environment, maybe a termination wrongly set. If you can, please share the environement so we can take a look together.

**Comment by Edwardle06** [comment]
[tracking_env_cfg.py](https://github.com/user-attachments/files/24125095/tracking_env_cfg.py)

[env_cfgs.py](https://github.com/user-attachments/files/24125110/env_cfgs.py)

That is great! Thanks very much!
